tests/foreman/cli/test_docker.py

```diff
@@ -1815,6 +1815,137 @@ def test_positive_container_admin_end_to_end_search(self):
+        """
+        pattern_prefix = gen_string('alpha', 5)
+        docker_upstream_name = 'alpine'
+        registry_name_pattern = (
+            "{}-<%= content_view.label %>/<%= repository.docker_upstream_name %>"
+        ).format(pattern_prefix)
+
+        # Satellite setup: create product and add Docker repository;
+        # create content view and add Docker repository;
+        # create lifecycle environment and promote content view to it
+        product = make_product_wait({'organization-id': self.org['id']})
+        repo = _make_docker_repo(
+                product['id'], upstream_name=docker_upstream_name)
+        Repository.synchronize({'id': repo['id']})
+        content_view = make_content_view({
+            'composite': False,
+            'organization-id': self.org['id'],
+        })
+        ContentView.add_repository({
+            'id': content_view['id'],
+            'repository-id': repo['id'],
+        })
+        ContentView.publish({'id': content_view['id']})
+        content_view = ContentView.info({
+            'id': content_view['id'],
+        })
+        lce = make_lifecycle_environment({'organization-id': self.org['id']})
+        ContentView.version_promote({
+            'id': content_view['versions'][0]['id'],
+            'to-lifecycle-environment-id': lce['id'],
+        })
+        LifecycleEnvironment.update({
+            'registry-name-pattern': registry_name_pattern,
+            'registry-unauthenticated-pull': 'false',
+            'id': lce['id'],
+            'organization-id': self.org['id'],
+        })
+        docker_repo_uri = " {}/{}-{}/{} ".format(
+                settings.server.hostname,
+                pattern_prefix,
+                content_view['label'],
+                docker_upstream_name
+        ).lower()
+
+        # 3. Try to pull in docker image from Satellite
+        remote_pull_command = 'docker pull {0}'.format(docker_repo_uri)
+        result = ssh.command(
+                remote_pull_command,
+                hostname=self.docker_host.ip_addr
+        )
+        self.assertEqual(result.return_code, 1)
+
+        # 4. Use Docker client to login to Satellite docker hub
+        result = ssh.command(
+                'docker login -u {} -p {} {}'.format(
+                        settings.server.admin_username,
+                        settings.server.admin_password,
+                        settings.server.hostname
+                ),
+                hostname=self.docker_host.ip_addr
+        )
+        self.assertEqual(result.return_code, 0)
+
+        # 5. Pull in docker image
+        # publishing takes few seconds sometimes
+        retries = 20 #if bz_bug_is_open(1452149) else 1
+        for i in range(retries):
+            result = ssh.command(
+                    remote_pull_command,
+                    hostname=self.docker_host.ip_addr
+            )
+            if result.return_code == 0:
+                break
+            sleep(2)
```

Comment:
Could have used the wait_for similar to https://github.com/omkarkhatavkar/robottelo/blob/892f122a1258d27c740f088eb8f4ce4d1d597134/tests/foreman/cli/test_repository.py#L134

Reply:
We can, although we don't use wait_for in this particular file, and another case there does use the same idiom with sleep.

Let's leave it as is and switch both these tests to `wait_for` in separate PR. How does it sound?

Reply:
Ok

Reply:
On other hand, we can do that in scope of current PR =) Static sleeps and loops like this are bad style. We are trying to follow only good practice in the code, so if have issue in it, it doesn't make sense to submit bad code and then change it in all places